types/hashing.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+import "github.com/minio/sha256-simd"
+
+func Hash(object Serializable) ([32]byte, error) {
```

Comment:
Could be done in a followup PR but I think this should be moved out of the types package. At least the way I currently understand the types package is: "blockchain core data types" similar to in the tendermint repo. 

We can merge this as is and rethink the repo structure as we make progress.